BuildScript/BuildTasks/MetadataValueDictionary.cs

```diff
@@ -32,17 +32,11 @@ public MetadataValueDictionary (IDictionary<string, string> rawDictionary)
       _rawDictionary = new Dictionary<string, string> (rawDictionary, StringComparer.OrdinalIgnoreCase);
     }
 
-    public string GetValue (IEnumerable<string> keys, Func<Exception> createEmptySequenceException)
+    public KeyValuePair<string, string> GetKeyValuePair (IEnumerable<string> possibleKeys, Func<Exception> createEmptySequenceException)
```

Comment:
The name needs work. It's not obvious that using multiple keys will result in exactly one match and an exception otherwise.

Reply:
You're right. In that case I'd just call it `Single`, analogous to what happens when calling `Single` on a usual `Dictionary`

Reply:
Sounds okay, am looking forward to actually seeing this so I can touch it.